admin/class-acc_user_importer-admin.php

```diff
@@ -373,14 +388,25 @@ private function proccess_user_data ( $users ) {
 				if (!empty($updatedFields)) {
 					// Passing in the $existingUser object with the updated values will persist everything to the database.
 					$updateResp = wp_update_user($existingUser);
-					if ( is_wp_error($existingUser) ) {
+					if ( is_wp_error($updateResp) ) {
 						$this->log_dual(" > failed to update user");
 						$this->log_dual(" > WP:" . $updateResp->get_error_message());
 						continue;
 					}
-
-					$updated_users[] = $userContactId;
+					$this->log_dual(" > updated user");
+					$updated_users[] = $accUserData['display_name'];
 					$updated_users_email[] = $userEmail;
+
+					//user_login is a special case, not updated by wp_update_user.
+					//If this field changed, use a SQL command to update it.
+					//See https://wordpress.stackexchange.com/questions/103504/how-to-programatically-change-username-user-login
+					if (in_array('user_login', $updatedFields)) {
```

Comment:
To be honest, I am not sure if this is a good idea. With this, we are bypassing some safety guards put in place by Wordpress and some other plugins might have the assumption that a username never changes and break in the process. Our two systems might be fine right now, but it might bight a poor ACC webmaster in the future :sweat_smile: 
This logic and the code block above handling family memberships are only required to support a soft migration from the name-based `login_name` to the Contact ID based `login_name`. I am a bit hesitant making that a core functionality of a public plugin and rather focus on one unique identifier and implement it cleanly. That said, reading the email chain with Jeffrey, things we assumed are unique might not be :grimacing:
I'd prefer to wait until past the meeting tomorrow and then discuss how to proceed. It might make sense to provide an ID migration functionality as part of this plugin (e.g. [extend the CLI](https://make.wordpress.org/cli/handbook/guides/commands-cookbook/)) vs handling everything in the import.